staging/src/k8s.io/apimachinery/go.mod

```diff
@@ -2,7 +2,7 @@
 
 module k8s.io/apimachinery
 
-go 1.24.0
+go 1.23.0
```

Comment:
I'm not sure I understand this ... this is saying `go 1.23` is required to build, but we want go 1.24 defaults for compat switches for runtime behavior?

We also unit test with go 1.24 only, so I really wouldn't want someone depending on one of our staging modules with significant implementations building with older go versions.

Things like cri-api / externaljwt / kms which are grpc types-only modules are more reasonable to leave at a backlevel, but not implementation modules like apimachinery, code-generator, and mount-utils

Reply:
@liggitt we could special case some?

Reply:
> we could special case some?

Though at what point do we just rip those out of staging and develop and test them as libraries?

(Also emphasis on the latter, they should have their own tests, purely type definition libraries could get away with minimal CI jobs for fmt/linting)